### ISSUE TITLE
solved

### ISSUE THREAD

**Issue by muses0229** [autonomousvision/LaRa#17]

**Comment by apchenstu** [comment]
It looks like a resolution issue, have you adjusted the bounding box [here](https://github.com/autonomousvision/LaRa/blob/main/lightning/network.py#L291) to your dataset?

**Comment by yejr0229** [comment]
No, it's still 0.5 in my setting, and I use the same images come from gobjaverse to train the model.

**Comment by apchenstu** [comment]
Do you have this issue if you do an inference using our pre-train model?

**Comment by yejr0229** [comment]
No, it works fine.

**Comment by apchenstu** [comment]
I noticed the "step_3540" indicator. Does this mean the model has only been trained for 3,540 steps so far? If so, you may need to continue training, as the normal-depth consistent loss likely hasn't started to take effect yet.

**Comment by yejr0229** [comment]
The multi-line image at the bottom indeed corresponds to the training step size in the name, but the teddy bear image above was obtained by training for more than 10 epochs for testing. At this time, the regularization terms should have been fully supervised.
![企业微信截图_17319337522007](https://github.com/user-attachments/assets/bd8378e6-ed56-42ca-8a97-bbd88354ee97)

**Comment by yejr0229** [comment]
And during the training process, I found that I need to add mask loss to eliminate redundant point clouds other than foreground objects. Otherwise, the reconstruction result will always contain a large number of point clouds that should not appear. Are these two types of correlation?
![image](https://github.com/user-attachments/assets/00ba1fe2-012d-4c98-bc16-6bf96c870d80)

**Comment by apchenstu** [comment]
I see, I didn't have this issue but looks like you are not using random background [here](https://github.com/autonomousvision/LaRa/blob/main/dataLoader/gobjverse.py#L106), which is supposed to be used during training. Any modification to the original code and dataset? How about the batch size and number of GPUs?

**Comment by apchenstu** [comment]
Wow, 4-8 h100;) What does pbr image mean? Supervising albedo, roughness and metals at the same time?

**Comment by yejr0229** [comment]
Yes, I have try two version. First is just simply change 'tar_img' to 'albedo'/'roughness'/'metallic' and training a lara for reconstructing a/r/m input. And the second one is like you said, let one GS point carry three PBR components and supervising at the same time. In my setting, the input to the reconstructed model is PBR images, and Lara is trained with RGB images, its performance drops more when reconstructing them. so I have to train a new lara for my PBR input.

**Comment by apchenstu** [comment]
> Hi, sorry for bothering again, I visualized surf_normal and render_normal in lara training. This is very different from what is shown in the 2dgs paper. Is this the way it is or is it because of other problems? Thanks in advance! ![image](https://private-user-images.githubusercontent.com/132911839/391083289-c52f72ac-de59-4303-8075-63fc2ecd1ae9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI5MTYxNjMsIm5iZiI6MTczMjkxNTg2MywicGF0aCI6Ii8xMzI5MTE4MzkvMzkxMDgzMjg5LWM1MmY3MmFjLWRlNTktNDMwMy04MDc1LTYzZmMyZWNkMWFlOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyOVQyMTMxMDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNjY5ZDU3ODhjN2I2NTVkNGQ4NDFhNTkxYjgyMzBjODk1ZjQ5NDE3Y2I5MjQ2Zjc4OGY4NWI2NjdmOWJhNzNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.wf7qkPmeJtB5cjs5vOeXsNOHTeLGGOTaM1g4D9tPDaU)

The normal map looks like haven't been normalized and clamped

**Comment by yejr0229** [comment]
Thanks for the help! I have successfully changed the direction of the normal and transformed it into object space.
![企业微信截图_17329352533584](https://github.com/user-attachments/assets/5e492bf5-54f4-4a78-8073-f0b977b40431)

**Comment by apchenstu** [comment]
> Thanks for the help! I have successfully changed the direction of the normal and transformed it into object space. ![企业微信截图_17329352533584](https://private-user-images.githubusercontent.com/132911839/391222484-5e492bf5-54f4-4a78-8073-f0b977b40431.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI5NTUwMzcsIm5iZiI6MTczMjk1NDczNywicGF0aCI6Ii8xMzI5MTE4MzkvMzkxMjIyNDg0LTVlNDkyYmY1LTU0ZjQtNGE3OC04MDczLWYwYjk3N2I0MDQzMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEzMFQwODE4NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMzI4NzM1MWY2M2VkNTU3ZWVmYWIyZTQ3MzVmMzY5YmZkNzY1MTc4MWUyNjdmYjI1ZDkxMWIzZTBhMDYyYTVlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.aD2pYGDWEW13G4coQpdPufHrt6PBRFgw7o52-ceJ1D0) I would like to ask again what is the difference between surf_normal and render_normal？

Good to see you figure it out, the surf_normal is the derived normal using depth, and render_normal is the rendered by volume integration (similar to RGB integration).

**Comment by yejr0229** [comment]
So can I understand it this way, rend_normal is the normal obtained by 2dgs when rendering in cuda, which is the true value of the normal, while surf_normal is obtained by the cross product of the two components dx and dy of scaling, which is a pseudo-true value, the Regularization loss is to supervise surf_normal and make it as close as possible to rend_normal, right?
